### ISSUE TITLE
Investigate "Error 103" on one drive integration

### ISSUE THREAD

**Issue by marcospri** [hypothesis/lms#3203]
We have seen an "Error 103" pop up intermittently on the one drive integration

After successfully authorizing with a one drive account the one drive modal shows a page with an "Error 103" message an button to continue. After continuing the modal looks exactly like the file picker one but you are not picking a file but browsing them, there's no "select" button and it's not very apparent what's going on.

What we know so far:

- You can reproduce forcing one drive to ask for a password: before using the filepicker go to https://account.microsoft.com/account/Account and in the top right corner click sign out or "sign out and forget" in the three dots near to the account name.

- The error itself seems to be exactly this one: https://github.com/OneDrive/onedrive-api-docs/issues/741 (the only time I've seen the error I had devtools open and the errors/messages matched those)

   That means that somehow the API request that the filepicker is doing on startup are failing. Note that his happens on the window the file picker opens, outside of the LMS's app control.

- It looks like once you get the error,  closing the picker window and starting over "fixes" it.

- ~It seems to only happen the first time an user access the integration. It doesn't happen always (or often) when that's the case.~

- ~To reproduce that situation (which again, might not be the cause) you can go to https://account.live.com/consent/Manage?uaid=db10ff0f8f2e45a2bd2e6380faad093d and remove access to the hypothesis (qa/prod/dev) app.~

- ~Might be unrelated but the picker docs state here https://docs.microsoft.com/en-us/onedrive/developer/controls/file-pickers/js-v72/?view=odsp-graph-online#setting-up that we need to `Click Generate New Password to create an Application secret. While this value is not necessary for the picker, it must have been created.` that's has been done after seeing this error, it's unclear if it helped as is not easy to reproduce the error.~

**Comment by marcospri** [comment]
Created issue at MS repo: https://github.com/OneDrive/onedrive-api-docs/issues/1541#issuecomment-939945255

**Comment by marcospri** [comment]
Closing this without finding a solution. I don't think there's a problem in the code we use for this (which is not much) and I  thought the problem would be on the "app registration" side but I changed every option there that made sense to change and many that didn't and I still get the same result.

Also:

- The problem happens with OneDrive accounts but no SharePoint ones.
- Other services with a similar integration (eg Asana) seem to have the exact same problem.
